### PR TITLE
Replace size() with empty() whenever possible

### PR DESCRIPTION
**Pull request by brunoabinader**
This is good because `empty()` is O(1) while `size()` _might be - depends on platform and/or compiler version -_ O(n).

### CODE REVIEW

File: src/clipper/clipper.cpp [hunk 1]
```diff
@@ -2013,7 +2013,7 @@ void Clipper::InsertLocalMinimaIntoAEL(const cInt botY)
 
     //if any output polygons share an edge, they'll need joining later ...
     if (Op1 && IsHorizontal(*rb) && 
-      m_GhostJoins.size() > 0 && (rb->WindDelta != 0))
+      !m_GhostJoins.empty() && (rb->WindDelta != 0))
```
Comment (1ec5) [inline src/clipper/clipper.cpp]: This file comes from the [polyclipping](https://sourceforge.net/p/polyclipping/) library. In the past, we’ve [attempted](https://github.com/mapbox/mapbox-gl-native/pull/1425#commitcomment-11049117) to keep this file in sync with the upstream project.

Comment (brunoabinader) [inline src/clipper/clipper.cpp]: Ah, cool - thanks for noticing :-) I'll send a patch upstream.

**Comment by ljbade** [comment]
@kkaefer Should we merge this?

**Comment by kkaefer** [comment]
Yes, this should be merged.

**Comment by brunoabinader** [comment]
> Yes, this should be merged.

I am rebasing & updating the patch, then will proceed with merging it, thanks! :+1:

**Pull request merged**